### PR TITLE
Test jprod and jtprod in SlackModel

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

**Comment by dpo** [comment]
Testing uncovered a number of bugs here!

File: src/jump_model.jl [hunk 1]
```diff
@@ -197,7 +197,7 @@ function jtprod!(nlp :: JuMPNLPModel,
                 Jtv :: Array{Float64})
   nlp.counters.neval_jac -= 1
   nlp.counters.neval_jtprod += 1
-  Jtv[:] = jac(nlp, x)' * v
+  Jtv[1:nlp.meta.nvar] = jac(nlp, x)' * v
```
Comment (dpo) [inline src/jump_model.jl]: I'm not sure why this doesn't bite us with `AmplModel`s.

Comment (dpo) [inline src/jump_model.jl]: Oh well, it probably does. It's just that `SlackModel(AmplModel)` isn't tested.

**Comment by abelsiqueira** [comment]
Yikes, I completely missed [this line](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/develop/src/slack_model.jl#L45). Slack model is completely untested in terms of consistency. I'll merge this and see what I can do.

**Comment by dpo** [comment]
Thanks!

**Pull request merged**